### DISCUSSION TITLE
Opening multiple of the same HID devices

### DISCUSSION THREAD

**Discussion by Edmonds1997** [libusb/hidapi#678] [Q&A] [answered]
Hello, I am trying to use the HID API to open some HID USB devices. Two of the devices are the same and have identical product ids, vendor ids, AND serial numbers. Is there anything I can do to get around this?

**Comment by Youw** [comment, marked as answer]
They are still going to be enumerated as different devices - different PATHs.

But it is all up to you. You should define yourself how are you going to distinguish the two, at first - logically.
If those are your devices (you have control over the FW) - preferably generate different SN for each.
If you don't have control of the FW but know the API/communication protocol/available reports/etc. - you could try to use those to identify which one is which.

There is not really a lot you can do here.
